### ISSUE TITLE
Ethereum validators CL and EL events should respect the ownership percentage

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#8438]
## Abstract

A user (@gnidan) requested this. At the moment rotki does not respect validator percentage when creating withdrawal events (CL rewards) or block/mev events (EL rewards).

Ideally rotki should check the validator ownership percentage and for each of those make sure that events count the reward with the given percentage.

What's more if the user edits the validator percentage all events should be recalculated.

## Task

- See how hard it would be to do
- Spec it out
- Maybe do it (update this section then)

**Comment by gnidan** [comment]
Possible bug that's related to this thematically: APR values are incorrect by a factor of ownership percentage.

I jointly own a validator (50%/50%), and I operate this validator through an account owned entirely by me. To address the accounting discrepancy, I have customized the events for the validator deposit and all block rewards+withdrawals to use 50% amounts (since I only paid half the deposit, and I only earn half of the rewards). Problem is, when I look at the APR on the staking dashboard, it appears to be computing the percentage based on _my share_ of the earnings (this is correct), but based on the _full joint investment_ of 32 ETH (this is not correct), so as a result, the APR is half of what it should be. This then lowers my average APR for all my validators, making it harder to get a proper sense for financial performance.

**Comment by LefterisJP** [comment]
> Possible bug that's related to this thematically: APR values are incorrect by a factor of ownership percentage.
> 
> I jointly own a validator (50%/50%), and I operate this validator through an account owned entirely by me. To address the accounting discrepancy, I have customized the events for the validator deposit and all block rewards+withdrawals to use 50% amounts (since I only paid half the deposit, and I only earn half of the rewards). Problem is, when I look at the APR on the staking dashboard, it appears to be computing the percentage based on _my share_ of the earnings (this is correct), but based on the _full joint investment_ of 32 ETH (this is not correct), so as a result, the APR is half of what it should be. This then lowers my average APR for all my validators, making it harder to get a proper sense for financial performance.

this is for the performance in the eth staking view right? And yes there it's not take into account either and could also be done.

**Comment by gnidan** [comment]
> this is for the performance in the eth staking view right?

That's right - the "Staking per Validator" table has the wrong APR for the shared validator, and the "Staking / ETH" Overview panel also uses the incorrect APR value.

**Comment by gnidan** [comment]
Another thought of how to do this accounting is to make the events have a debt component and an earning component... so when I get a block reward, my account gets all the funds, but it's also marked with a "Generate Debt" event... then when I disburse the funds that don't belong to me, I can just mark it as "Repay Debt"
